### PR TITLE
Mock API

### PR DESCRIPTION
**Pull request by adscib**
As described in #30. I opted to use an environment variable to control whether the mock API is used and for now I'm just returning zeros or current state deterministically everywhere. I also updated the Demo-Drive notebook to use the mock API when no key is specified.

### CODE REVIEW

File: examples/Demo_Drive.py [hunk 1]
```diff
@@ -21,7 +21,12 @@
 parser.add_argument("--location", type=str, default="iai:ubc_roundabout")
 args = parser.parse_args()
 
-iai.add_apikey("")
+api_key = args.api_key or os.environ.get('IAI_API_KEY', None)
```
Comment (Ruishenl) [inline examples/Demo_Drive.py:24]: .get() will default to None

File: invertedai/api_resources.py [hunk 2]
```diff
@@ -30,6 +34,10 @@
 
 TIMEOUT = 10
 
+mock_api = bool(os.environ.get("IAI_MOCK_API", False))
```
Comment (Ruishenl) [inline invertedai/api_resources.py:37]: small suggestions.. strtobool is available to support slightly more cases
```
from distutils.util import strtobool
strtobool(os.environ.get("IAI_MOCK_API", "false")
```

True values are y, yes, t, true, on and 1; false values are n, no, f, false, off and 0

File: examples/Demo_Drive.py (hunk 1 above)
Comment (adscib) [reply examples/Demo_Drive.py:24]: I like to be explicit anyway.

File: invertedai/api_resources.py (hunk 2 above)
Comment (adscib) [reply invertedai/api_resources.py]: done

File: invertedai/utils.py [hunk 3]
```diff
@@ -33,6 +33,11 @@ def add_apikey(self, api_token: str = ""):
             raise error.InvalidAPIKeyError("Empty API key received.")
         self.session.auth = APITokenAuth(api_token)
 
+    def use_mock_api(self, use_mock: bool = True) -> None:
+        api_resources.mock_api = use_mock
+        if use_mock:
+            print('Using mock Inverted AI API - predictions will be trivial')  # TODO: replace with a suitable logger
```
Comment (adscib) [inline invertedai/utils.py:39]: Do we have some uniform mechanism in place to issue warnings like this?

Comment (Ruishenl) [reply invertedai/utils.py:39]: iai.logger

Comment (adscib) [reply invertedai/utils.py:39]: Is that safe in terms of circular imports?

Comment (Ruishenl) [reply invertedai/utils.py:39]: i think all the current reference of iai.logger happens inside the function so i never ran into any issue. We are already using it in this module

Comment (adscib) [reply invertedai/utils.py]: OK. Did I use it correctly or should I do some extra formatting?

Comment (Ruishenl) [reply invertedai/utils.py]: looks good to me

**Review by Ruishenl — APPROVED** [review]

**Comment by Ruishenl** [comment]
Feel free to merge once @AlirezaMorsali  is okay with it

**Comment by AlirezaMorsali** [comment]
I can fix the conflicts if you are working on something else.

**Comment by adscib** [comment]
Go ahead - I'm working on the docs.

**Pull request merged**